### DISCUSSION TITLE
Can cue validate a string‘s encoding type is utf8 or other?

### DISCUSSION THREAD

**Discussion by chenlinli** [cue-lang/cue#2065] [Q&A] [answered]
I want to validate a string's encoding type by cue? Like utf8 or not

**Comment by mvdan** [comment, marked as answer]
From https://cuelang.org/docs/references/spec/:

> String literals may only be valid UTF-8. Byte sequences may contain any sequence of bytes.

So string values in CUE, much like in Go, are always UTF-8.
